packages/server-core/src/user/identity-provider/identity-provider.hooks.ts

```diff
@@ -108,27 +150,52 @@ async function checkOnlyIdentityProvider(context: HookContext<IdentityProviderSe
 
 async function validateAuthParams(context: HookContext<IdentityProviderService>) {
   let userId = (context.actualData as IdentityProviderData).userId
+  let existingUser
 
-  if (context.params.authentication) {
-    const authResult = await context.app.service('authentication').strategies.jwt.authenticate!(
-      { accessToken: context.params.authentication.accessToken },
-      {}
-    )
-    userId = userId || authResult[appConfig.authentication.entity]?.userId
+  try {
+    if (userId) existingUser = await context.app.service(userPath).get(userId)
+  } catch (err) {
+    //
   }
 
-  if (!userId) {
-    if ((context.actualData as IdentityProviderData).type === 'guest') {
-      return
+  let isAuthenticated = false
```

Comment:
Just wondering if the code is similar to line 84 of same file. Can it be moved to a common method so that both code blocks can call it? Just a suggestion.